lib/hanami/action/params.rb

```diff
@@ -15,7 +13,13 @@ class Action
     #
     # @since 0.1.0
     class Params < BaseParams
-      include Hanami::Validations::Form
+      # @since 2.2.0
+      # @api private
+      class Validator < Dry::Validation::Contract
+        params do
+          optional(:_csrf_token).filled(:string)
```

Comment:
should this be `required` if `include Hanami::Action::CSRFProtection` is active?

Reply:
The CSRF class will fail the request if the `_csrf_token` parameter is missing, regardless of whether it's required here or not.

It's optional because in specs (or rather, `Hanami.env?(:test)`) `CSRFProtection` is a noop, and if you make it required then you have to pass in `_csrf_token` to every action spec.

Reply:
This current incantation has been in place [since 2.0.0.alpha1](https://github.com/hanami/validations/blob/v2.0.0.alpha1/lib/hanami/validations/form.rb#L44), and IMO, it's correct. 

It's not just about testing, either. Even when CSRF protection is enabled, not every action with a params schema/contract will be handling HTML form submissions. They could be handling any kinds of params (consider even GET requests with e.g. `?search=` param!). In these cases, the `_csrf_token` key should be allowed to be missing. If it is present, however, then we expect it to include a properly filled string.